src/minimizer.rs

```diff
@@ -1,5 +1,42 @@
-
+//! This module provides the base framework for all minimizers present in this crate, such as the
+//! base trait and return type.
 use ndarray::prelude::*;
+use std::time::Duration;
+
+/// Minimizer states at the end of the run
+#[derive(Debug, PartialEq)]
+pub enum RunStatus {
```

Comment:
Agreed, will change this. The PR is only a WIP for now anyway. I was thinking of adding more states, such as `warmup`, `inprogress`, one for various convergences. This can also be kept in the minimizer state (and updated via something like `is_finished` in NM). Maybe at some point it can also be potentially used to save the state of an in-progress minimization to restart or continue.